db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -290,7 +290,10 @@ private void messageLoop() {
     private void closeConnection() {
         if (this.channel != null) {
             try {
-                this.channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+                boolean terminated = this.channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
HI @thefringeninja! Yes, 10seconds would be two much, but it should not happen in the real world. We give 5 seconds of grace period for completing existing GRPC commands, then we force the shutdown, which will occur immediately and should only take a few milliseconds as GRPC library will directly kill all operations.

We could definitely reduce the second `awaitTermination()` time out in the code for the force shutdown (e.g. 1-2 secs), but in reality it will "feel" the same, as we will never reach the timeout.